src/components/FeatureViews/Courses/AttendanceTable.js

```diff
@@ -0,0 +1,437 @@
+    .filter((student) =>
+      sortByAlphabet === "asc" ||
+      sortByAlphabet === "desc" ||
+      sortByAlphabet === ""
+        ? student
+        : {
+            [student.studentName]: student.studentName,
+          }[sortByAlphabet]
+    )
+    .sort(
+      (firstEl, secondEl) =>
+        ({
+          asc: sortAscOrder(firstEl.studentName, secondEl.studentName),
+          desc: sortDescOrder(firstEl.studentName, secondEl.studentName),
+        }[sortByAlphabet])
+    );
+
+  return (
+    <TableContainer className={classes.table}>
+      <Table aria-label='simple table'>
+        <TableHead>
+          <TableRow>
+            <TableCell style={{ color: "black" }}>
+              <StudentFilterOrSortDropdown
+                students={studentsFullNameList}
+                setSortByAlphabet={setSortByAlphabet}
+                sortByAlphabet={sortByAlphabet}
+              />
+            </TableCell>
+            {sessions
+              .sort((firstSession, secondSession) => firstSession.id - secondSession.id)
+              .map(({ startDatetime, id: sessionId }, index) => (
+                <TableCell className={classes.tableCell}>
+                  {`Session ${index + 1} (${moment(startDatetime).format("MM/DD/YYYY")})`}
+                  {checkEditState(sessionId) ? (
+                    <Checkbox
+                      checked={isCheckBoxEditing[sessionId]}
+                      onChange={handleEdit}
+                      checkedIcon={
+                        <Check fontSize='small' style={{ color: "white" }} />
+                      }
+                      icon={<Add fontSize='small' />}
+                      id={sessionId}
+                      disableRipple
+                      classes={{
+                        root: classes.checkbox,
+                        checked: classes.checkbox2,
+                      }}
+                    />
+                  ) : (
+                    <SessionDropdownButton
+                      id={sessionId}
+                      attendanceEditStates={attendanceEditStates}
+                      setAttendanceEditStates={setAttendanceEditStates}
+                      courseAttendanceMatrix={courseAttendanceMatrix}
+                      setCourseAttendanceMatrix={setCourseAttendanceMatrix}
+                      setSortByAlphabet={setSortByAlphabet}
+                      index={index}
+                    />
+                  )}
+                </TableCell>
+              ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {studentAttendanceDisplay.map((row, studentIndex) => (
+            <TableRow key={row.studentId}>
+              <TableCell component='th' scope='row'>
+                {row.studentName}
+              </TableCell>
+              {row.attendanceList.map((sessionColumn, attendanceIndex) => (
+                <TableCell align='right' id={row.studentId}>
+                  {!checkEditState2(sessionColumn.sessionId) ?
```

Comment:
rename "checkEditState2" booleans should never be named as verbs like this. should be answerable as a true or false (i.e. "isEditState2Active") but not sure what is "EditState2" so this is confusing as well